### PR TITLE
DAOSSDE-58 SDL: T295: Avoid storing unencrypted confidential data without access control mechanism

### PR DESCRIPTION
**Pull request by GitHuaKuang**
Tighten the accessw control for some possible confidential data in log file and pool.

Change-Id: I175fca65f887695b5dac6298d8c27c6029d07d7d
Signed-off-by: Hua Kuang <hua.kuang@intel.com>

### CODE REVIEW

File: src/gurt/dlog.c
Comment (bfaccini) [inline src/gurt/dlog.c:677]: Why do you keep ug+r ? Also, what is the JIRA ticket associated with this PR ?

Comment (GitHuaKuang) [reply src/gurt/dlog.c:677]: It's a client log file, group user and others are able to read it given that only the owner can write it and no confidential data in the log. We had some discussion last couple of days in ESAD build/test forum.

https://jira.hpdd.intel.com/browse/DAOSSDE-58

**Review by NiuYawei — DISMISSED** [review]

**Review by cibervicho — APPROVED** [review]

**Comment by GitHuaKuang** [comment]
The failure is reported in https://jira.hpdd.intel.com/browse/DAOS-5587
@daos-stack/daos-gatekeeper can we land this PR?

**Comment by GitHuaKuang** [comment]
@daos-stack/daos-gatekeeper reminder of this PR is waiting for landing.

**Comment by johannlombardi** [comment]
99 failures?

File: src/gurt/dlog.c
Comment (bfaccini) [reply src/gurt/dlog.c:677]: hmmm, the ticket link is still unreachable for me (infamous "404 - Not Found" page) :-(
I remember that we have been talking about the different way this issue should be addressed to get PR reviewers also get access to associated JIRA, but is there any process in place now ? Again, I remember that it could be possible to define groups (users, developers,...) of people who can get access, like having all DAOS teammates to be put in developer group so they could be the only to get access to some tickets...

**Review by bfaccini — APPROVED** [review]

**Comment by jolivier23** [comment]
> @johannlombardi "99 failures?" Can you have more details of what's this 99 failures? CI test failure, or confidential data handling failure?

99 tests failed in the results

> @johannlombardi "99 failures?" Can you have more details of what's this 99 failures? CI test failure, or confidential data handling failure?

Actual 128 failures: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3341/4/tests

Can you merge latest master just to make sure your permissions change didn't introduce anything?

**Comment by GitHuaKuang** [comment]
Oh, sure.
Only 1 failure on previous run:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3341/3/tests

After merge to the latest,  just like the 3rd tun, the Github showed all tests on GREEN except continuous-integration/jenkins/pr-head. Thought it's the same 1 failure, guess I didn't check the Jenkins report. Rebasing it again, will be the 4th consecutive merge.

**Comment by jolivier23** [comment]
> Oh, sure.
> Only 1 failure on previous run:
> https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3341/3/tests
> 
> After merge to the latest, just like the 3rd tun, the Github showed all tests on GREEN except continuous-integration/jenkins/pr-head. Thought it's the same 1 failure, guess I didn't check the Jenkins report. Rebasing it again, will be the 4th consecutive merge.

It's expected that only PR-head is red.   Green on the test runs just means they ran, not that there are no test failures.   You have to look at the report to see why pr-head is red

**Comment by GitHuaKuang** [comment]
@daos-stack/daos-gatekeeper Kept merging several times, like master CI, every time this PR had different set of failures, this time still seeing 14 existing failures, look similar to the failures of some landed PRs, not sure we should keep waiting till the master passed its own CI?

**Comment by mjmac** [comment]
> 
> 
> @daos-stack/daos-gatekeeper Kept merging several times, like master CI, every time this PR had different set of failures, this time still seeing 14 existing failures, look similar to the failures of some landed PRs, not sure we should keep waiting till the master passed its own CI?

It's not enough to just say "well, looks like the same number of failures as on master" and ask for landing. The expectation is that engineers asking for an exception to land with failures will have looked at each failure and matched it with an existing ticket. This discipline ensures that we don't accidentally introduce new regressions and prolong the misery of instability, and also allows management to prioritize failing test tickets based on frequency. It's annoying and tedious, but someone has to do it.

**Comment by johannlombardi** [comment]
Agreed with @mjmac. Please have a look at the failure and report back.

**Comment by GitHuaKuang** [comment]
Added this 4 failures case to DAOS-5610 which have these failures reported already.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Can't land with 66 failures given current state of master is much better.  Please merge latest again

**Comment by GitHuaKuang** [comment]
Appears like only one failure which is Zero-config test reported by DAOS-5853

**Review by cibervicho — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**
